<!doctype html>
<html lang="ko">
<head>
    <meta charset="utf-8">
    <style>
    </style>
    
</head>

<body>
    <h1 onclick="eventHandler();">예외처리</h1>
    <h2>try, catch, finally</h2>
    <p>try: 정상적으로 실행되는 경우 실행될 프로그램 작성<br>
       try 블록안에서 에러가 발생한 경우 catch 블록으로 이동함</p>
    <p>catch:  try 블록에서 에러가 발생한 경우<br>
        에러를 인자 e로 받아 에러를 처리하는 코드를 작성</p>
    <p>finally: try, catch구문이 실행되고 나서 실행될 코드</p>
    <h2>finally가 없어도 똑같이 실행이 되지만 만약 catch가 없을경우 시멘틱에러가 발생함</h2>
    
<script type="text/javascript">
    function errFunc(){
        throw "error";
        console.log("this code will not be excuted");
    }
    function func(){
        try{
        console.log("function - 1");
        errFunc();
        console.log("function - 2");
        }
        catch(e){
            console.log("catch error in function : ",e);
        }
        finally{
            console.log("finally in function - this code will always be excused");
        }

    }
    try{
        console.log("try - 1");
        func();
        console.log("try - 2");
    }
    catch(e){
        console.log("catch error : ",e);
    }
    finally{
        console.log("finally - this code will always be excused");
    }

</script>
</body>

<footer>
</footer>
</html>